src/main/scala/com/fulcrumgenomics/umi/ReviewConsensusVariants.scala

```diff
@@ -52,82 +129,197 @@ import scala.collection.mutable.ListBuffer
     def toQueryInterval: QueryInterval = new QueryInterval(dict.getSequenceIndex(chrom), start, start)
+    def toInterval: Interval = new Interval(chrom, start, start)
   }
 
   override def execute(): Unit = {
     def f(ext: String): Path = output.getParent.resolve(output.getFileName + ext)
+
     val consensusIn  = SamReaderFactory.make().open(consensusBam)
     val groupedIn    = SamReaderFactory.make().open(groupedBam)
     val factory      = new SAMFileWriterFactory().setCreateIndex(true)
     val consensusOut = factory.makeWriter(consensusIn.getFileHeader, true, f(".consensus.bam").toFile, ref.toFile)
-    val groupedOut   = factory.makeWriter(groupedIn.getFileHeader, true, f(".grouped.bam").toFile, ref.toFile)
+    val groupedOut   = factory.makeWriter(groupedIn.getFileHeader,   true, f(".grouped.bam").toFile,   ref.toFile)
 
     // Pull out the consensus reads
     logger.info(s"Loading variants from ${input.toAbsolutePath}")
     val variants        = loadPositions(path=input, refFile=refFile)
+    logger.info(s"Loaded ${variants.size} variant positions for review.")
     val variantsByChrom = variants.groupBy(v => v.chrom)
     val queries         = variants.map(_.toQueryInterval).toArray
     val sources         = mutable.HashSet[String]()
 
     logger.info(s"Extracting consensus reads from ${consensusBam.toAbsolutePath}")
     consensusIn.query(queries, false).filter(r => nonReferenceAtAnyVariant(r, variantsByChrom)).foreach { rec =>
       consensusOut.addAlignment(rec)
-      Option(rec.getStringAttribute(tag)).map(mi => mi.takeWhile(_ != '/')).foreach(sources.add)
+      sources.add(toMi(rec))
     }
     consensusIn.safelyClose()
     consensusOut.close()
 
     // And now the raw reads
     logger.info(s"Extracting raw/grouped reads from ${groupedBam.toAbsolutePath}")
     groupedIn.query(queries, false)
-      .filter(rec => Option(rec.getStringAttribute(tag)).map(mi => mi.takeWhile(_ != '/')).exists(sources.contains))
+      .filter(rec => sources.contains(toMi(rec)))
       .foreach(groupedOut.addAlignment)
     groupedIn.safelyClose()
     groupedOut.close()
+
+    // Generate the details output
+    logger.info("Generating review file.")
+    generateDetailsFile(consensusBam, f(".grouped.bam"), variants, f(".txt"))
+  }
+
+  /**
+    * Generates the detailed consensus review file.
+    *
+    * @param consensusBam the original consensus bam (not the review consensus bam)
+    * @param groupedBam the grouped bam (the review version)
+    * @param variants the set of variants to report on
+    * @param output the file to write
+    */
+  private def generateDetailsFile(consensusBam: PathToBam, groupedBam: PathToBam, variants: Seq[Variant], output: FilePath): Unit = {
+    val consensusReader = SamReaderFactory.make().open(consensusBam)
+    val groupedReader   = SamReaderFactory.make().open(groupedBam)
```

Comment:
I guess?  CRAM still seems pretty broken to me and I'm not super keen on passing more stuff around to support CRAM until it actually works.

Perhaps what we should do is centralize all generation of readers/writers into methods in the `Bams` object so it's easy to track them down?